src/mds/SessionMap.h

```diff
@@ -594,7 +594,7 @@ class SessionMapStore {
 class SessionMap : public SessionMapStore {
 public:
   SessionMap() = delete;
-  explicit SessionMap(MDSRank *m) : mds(m) {}
+  explicit SessionMap(MDSRank *m);
```

Comment:
can't we keep this intact and initialise `mds_session_metadata_threshold` here?

Reply:
Yes, that's possible, however, the _preferred_ approach is to keep ctor/dtor which are not no-op in cc source.